.github/workflows/tag_release.yaml

```diff
@@ -0,0 +1,75 @@
+name: "Create Tag on Pull Request merge"
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  TagMerge:
+    name: Tag on Pull Request merge
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    steps:
+      - name: Generate Tag
+        id: tagged
+        run: |
+          echo $(curl \
+            --silent \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            "https://api.github.com/repos/${{ github.repository }}/tags" \
+            | jq -r '.[].name |= sub("-rc0|v"; "";"g") | select(length > 1) | max_by(.name) | .name' \
+            | awk '
+            function inc(s){
+              split(s, a, ".")
+              a[3]++
+              if (a[3]>=10){
+                a[2]++;a[3]=0
+              }
+              if (a[2]>=10){
+                a[1]++;a[2]=0
+              }
+              return a[1]"."a[2]"."a[3]
+
+            }
+            { print "newtag="inc($1) }') >> $GITHUB_OUTPUT
+
+      - name: Create new tag in github
+        id: createtag
+        run: |
+          echo tag_sha=$(curl \
+            --silent \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            "https://api.github.com/repos/${{ github.repository }}/git/tags" \
+            -d '{"tag":"v${{ steps.tagged.outputs.newtag }}","message":"New release from github actions","object":"${{ github.sha }}","type":"commit"}' \
+            | jq -r '.sha') >> $GITHUB_OUTPUT
```

Comment:
I'm maybe missing something, but now we have the right tag, we can drop back to using `git` itself here right?

I.e. a simple `git tag ${{ steps.tagged.outputs.newtag }} && git push --tags` or so?

Reply:
I also tried this out, but it still requires you to download the repo; the image starts out blank and using the github api is the fastest way to handle it, imo.